pkg/io/serialize.go

```diff
@@ -31,7 +31,7 @@ var byteOrder = binary.LittleEndian
 
 // Encode encodes multiple primitive values into a writer.
 // All passed values must be copies, not references.
-func Encode(writer io.Writer, values ...interface{}) (err error) {
+func Encode(writer io.Writer, values ...interface{}) (err error) { //nolint: cyclop
```

Comment:
Usually try to avoid `//nolint`. If you need it, add a comment why. Why you need it here?

Reply:
It is needed here because, the Encode function by design has a long switch case with one case for each switch statement, which cause the cyclomatic complexity to exceed the limits in our linter.

Added a comment in 464caf77.